### ISSUE TITLE
objectMap some times  erace twice, cause a crash

### ISSUE THREAD

**Issue by waeilua** [bersler/OpenLogReplicator#31]
hi @bersler :
the objectMap some times  erace twice, cause a crash.
see the code comments

In Schema.cpp func 
void Schema::rebuildMaps(void);

```
for (auto it = objectMap.cbegin(); it != objectMap.cend() ; ) {

                OracleObject* object = it->second;
                if (object->user == user) {
                    removeFromDict(object); // here, objectMap erase object
                    INFO("dropped schema: " << object->owner << "." << object->name << " (dataobj: " << std::dec << object->dataObj
                            << ", obj: " << object->obj << ")");
                    objectMap.erase(it++);//if the it point to zhe removeFromDict() erased object, crash
                    delete object;
                } else {
                    ++it;
                }
            }
```
in func 
void Schema::removeFromDict(OracleObject* object)
```
    void Schema::removeFromDict(OracleObject* object) {
        if (objectMap.find(object->obj) == objectMap.end()) {
            CONFIG_FAIL("can't remove object (obj: " << std::dec << object->obj << ", dataObj: " << object->dataObj << ")");
        }
        objectMap.erase(object->obj);// erase object
```
in func 
void Schema::addToDict(OracleObject* object) 
```
        if (objectMap.find(object->obj) != objectMap.end()) {
            CONFIG_FAIL("can't add object (obj: " << dec << object->obj << ", dataObj: " << object->dataObj << ")");
        }
        objectMap[object->obj] = object;//add obj
```

**Comment by bersler** [comment]
Thank you for reporting.

**Comment by bersler** [comment]
Fixed in version 0.9.42
